src/tensorinfo.cpp

```diff
@@ -309,6 +309,7 @@ std::shared_ptr<TensorInfo> TensorInfo::createCopyWithNewShape(const shape_t& sh
 std::shared_ptr<TensorInfo> TensorInfo::createCopyWithEffectiveDimensionPrefix(size_t dim) const {
     auto copy = std::make_shared<TensorInfo>(*this);
     copy->influencedByDemultiplexer = true;
+    copy->effectiveShape = this->getEffectiveShape();
```

Comment:
Why this is required? Shouldn't that be handled in copy constructor in L310?

Reply:
No, because `getEffectiveShape` does not return always the same class member. Depending if `this.effectiveShape` is empty or not, it chooses to return `this.shape` instead.

In a scenario with custom node as entry, variable `this.effectiveShape` is empty, because effective shape is the same as `this.shape` (layout change does not work on custom libraries).

What will happen when you copy `this.effectiveShape` (which is empty) and add demultiplexer dimension? We will end up with one element - `(0)` (equal to demultiply count).

To summarize, this is required to first get effective shape from source object (which will return `this.shape`) and then add demultiplexer dimension afterwards. This way we can be sure to end up with correct effective shape.

Reply:
You're right. It was bug from the start then.

Reply:
Actually I see that all constructors initialize effectiveShape so we might want to redefine getEffectiveShape. Unless there is some special case